### ISSUE TITLE
Network/server errors no longer caught by feedparser

### ISSUE THREAD

**Issue by Safihre** [kurtmckee/feedparser#239]
This could be intended behavior, but in feedparser 6.0.0+ it no longer catches all the network/server errors that can happen.
I saw in the 5.x code that there is just a general `try/finally`, that's not there anymore in 6.x:
https://github.com/kurtmckee/feedparser/blob/consistent-summary-and-description/feedparser/feedparser.py#L2984-L2987

For example:
**feedparser 5:**
```python
import feedparser
feedparser.parse("http://nothing")
{'feed': {}, 'entries': [], 'bozo': 1, 'bozo_exception': URLError(gaierror(11001, 'getaddrinfo failed'))}
```
**feedparser 6.0.1:**
```python
import feedparser
feedparser.parse("http://nothing")
Traceback (most recent call last):
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\urllib\request.py", line 1350, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\http\client.py", line 1240, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\http\client.py", line 1286, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\http\client.py", line 1235, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\http\client.py", line 1006, in _send_output
    self.send(msg)
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\http\client.py", line 946, in send
    self.connect()
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\http\client.py", line 917, in connect
    self.sock = self._create_connection(
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\socket.py", line 787, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 11001] getaddrinfo failed
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "C:\Users\me\Documents\GitHub\sabnzbd3\venv\lib\site-packages\feedparser\api.py", line 214, in parse
    data = _open_resource(url_file_stream_or_string, etag, modified, agent, referrer, handlers, request_headers, result)
  File "C:\Users\me\Documents\GitHub\sabnzbd3\venv\lib\site-packages\feedparser\api.py", line 114, in _open_resource
    return http.get(url_file_stream_or_string, etag, modified, agent, referrer, handlers, request_headers, result)
  File "C:\Users\me\Documents\GitHub\sabnzbd3\venv\lib\site-packages\feedparser\http.py", line 158, in get
    f = opener.open(request)
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\urllib\request.py", line 525, in open
    response = self._open(req, data)
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\urllib\request.py", line 542, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\urllib\request.py", line 502, in _call_chain
    result = func(*args)
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\urllib\request.py", line 1379, in http_open
    return self.do_open(http.client.HTTPConnection, req)
  File "C:\Users\me\AppData\Local\Programs\Python\Python38\lib\urllib\request.py", line 1353, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 11001] getaddrinfo failed>
```

**Comment by thraxil** [comment]
I encountered this as well. The new behavior is fine, but this backwards-incompatible change really ought to have been documented. But there doesn't appear to be any changelog entries since 4.2...

**Comment by kurtmckee** [comment]
I don't recall deliberately introducing this crash behavior and deliberately choosing not to document it.

Perhaps this is just a bug that was introduced during refactoring and can be fixed with a patch.

**Comment by kurtmckee** [comment]
This should be fixed in feedparser 6.0.7. Please let me know if you encounter any other issues!
